src/get-eaw.ts

```diff
@@ -23,6 +23,36 @@ function getEAWOfCodePoint(codePoint: number): EastAsianWidth {
   return defs[min][2];
 }
 
+/**
+ * Gets the EAW property of a code point.
+ * @param codePoint Code point
+ * @return The EAW property of the code point
+ * @example
+ * import { getEAWOfCodePoint } from .codePointAt(0)"meaw";
+ *
+ * // Narrow
+ * assert(getEAWOfCodePoint("A".codePointAt(0)) === "Na");
+ * // Wide
+ * assert(getEAWOfCodePoint("あ".codePointAt(0)) === "W");
+ * assert(getEAWOfCodePoint("安".codePointAt(0)) === "W");
+ * assert(getEAWOfCodePoint("🍣".codePointAt(0)) === "W");
+ * // Fullwidth
+ * assert(getEAWOfCodePoint("Ａ".codePointAt(0)) === "F");
+ * // Halfwidth
+ * assert(getEAWOfCodePoint("ｱ".codePointAt(0)) === "H");
+ * // Ambiguous
+ * assert(getEAWOfCodePoint("∀".codePointAt(0)) === "A");
+ * assert(getEAWOfCodePoint("→".codePointAt(0)) === "A");
+ * assert(getEAWOfCodePoint("Ω".codePointAt(0)) === "A");
+ * assert(getEAWOfCodePoint("Я".codePointAt(0)) === "A");
+ * // Neutral
+ * assert(getEAWOfCodePoint("ℵ".codePointAt(0)) === "N");
+ */
+export function getEAWOfCodePoint(codePoint: number): EastAsianWidth | undefined {
+  if (!Number.isInteger(codePoint) || codePoint <= 0 || 0x10FFFF <= codePoint) return undefined;
```

Comment:
U+0000 and U+10FFFF are both valid code points.
Please fix the range like below, and also add test cases to ensure that 0 and 0x10FFFF are treated as valid.

```suggestion
  if (!Number.isInteger(codePoint) || codePoint < 0 || 0x10FFFF < codePoint) return undefined;
```